src/components/study-pane.js

```diff
@@ -999,9 +1019,30 @@ const StudyPane = (props) => {
                         deletedEquipmentId
                     );
                 }
+
+                // study partial update :
+                // loading equipments involved in the study modification and updating the network
+                const substationsIds =
+                    studyUpdatedForce.eventData.headers['substationsIds'];
+                const tmp = substationsIds.substring(
+                    1,
+                    substationsIds.length - 1
+                ); // removing square brackets
+                if (tmp && tmp.length > 0) {
+                    updateNetwork(tmp.split(', '));
+                }
             }
         }
-    }, [studyUpdatedForce, updateNetwork, removeEquipmentFromNetwork]);
+    }, [
+        history,
+        studyUuid,
+        studyUpdatedForce,
+        updateNetwork,
+        network,
```

Comment:
many more trigger, you don't want to trigger this one on
history & network 
You can use ref to avoid this like this
const network = useSelector(...)
const  networkRef = useRef();
networkRef.current = network;
this way the reference value never change but is always up to date, but do not trigger unwanted effect

Reply:
Done. We only trigger on the 3 hooks that were in dependance before.